releases/stable-branches-and-release-cycle.md

```diff
@@ -0,0 +1,79 @@
+# Producing and supporting stable releases
+
+## Summary
+
+Today the Submariner project releases one minor version a month without capacity
+for stable branches or bugfix releases. In the following proposal we define a plan
+to make this possible starting with 0.8.
+
+## Proposal
+
+### Alignment of branch -> image releases
+
+We produce currently `<image>:devel` for every master push, along with `<image>:x.y.z` for
+the releases.
+
+Changing the `master` branch name to `devel` would make this homogeneous so our image releases
+are simply `<image>:<branch-or-tag>`.
+
+### Creating one stable branch for every minor release
+
+Approaching the release of every minor or major version a new stable branch will be forked
+off `devel` branch in the form of `release-x.y` for every existing repository.
+
+Any changes related to pinning shipyard base image will be done in that branch.
+
+At that point in time, any patches which belong to the release will need to be merged in master
+and backported to `release-x.y`. For this reason, the later we branch, the less extra work
+will be necessary to produce a stable release.  
+
+### Maintaining a stable branch
+
+Any patches that are necessary for fixing bugs in a stable branch will need to be merged
+first into master, and the later stable branches.
+
+`release-0.8`  <-- `release-0.9` <-- `devel`
+
+Patches could be simple cherry-picks, or sometimes they could need specific adaptations
+to the branch.
+
+When a patch is merged to a stable branch the images for such branch are updated, for example
+`quay.io/submariner/submariner:release-0.9`
+
+### How long is a stable branch maintained
+
+Initially stable branches will be maintained until the next stable release is released,
+this would change in the future if we introduce the concept of long term support releases.
+
+### Reducing the cadence of minor releases
+
+Based on the previous ideas the existing minor version cadence doesn't seem realistic
+for maintaining stable branches. We don't want fake stable branches where patches
+are never backported.
+
+With the new cadence after each sprint we would produce a milestone release,
+for example
+
+* `0.9-milestone-1-rc0`, ... , `0.9-milestone-1`
+* `0.9-milestone-2-rc0`, ... , `0.9-milestone-2`
+* `0.9-milestone-3-rc0`, ... , `0.9-milestone-3`
+* `0.9-milestone-4-rc0`, ... , `0.9-milestone-4`
+
+In this case we won't do pinning of shipyard base image, but we will release and update
+the admiral & shipyard `go.mod` references.
+
+After a number of milestones instead we would follow the usual pattern for a release:
+
+* `0.9.0-rc0`, ..., `0.9.0` along with it's stable branch `release-0.9` as described
+previously.
+
+The actual cadence and release times is TBD by the community, it could be aligned with
+k8s for example, slightly after a release of k8s we could adapt to the new changes
+in k8s API and produce a final release.
```

Comment:
I think it makes sense to release once theres what to release (eg enough features).

Reply:
or sync with AMC

Reply:
We could make exceptions at some point, but I believe that setting a fixed cadence is good for projects, users will know what to expect, when to expect, we will know how many stable branches we need to maintain.

But as you say, at the start it's probably good to allow freedom to create an additional release, as long as we are aware of the cost resulting of that.